src/hooks/number-validator/number-validator.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
+
+const numberRegex = RegExp(/^-?(0|[1-9]\d*)?[,.]?(\d+)?$/);
```

Comment:
I'm not sure this is a blocker. But I was playing around with this and found this odd:

| number | valid? |
| --- | --- |
| 0 | true |
| 00 | true |
| 001 | true |
| 001. | false |
| 1. | true |

Very unexpected. But also unlikely we care. I think this is a kind of thing to throw to QA and product.

Reply:
I also found it odd that adding leading or trailing space breaks the input as well.

Reply:
Makes sense that space characters are invalid, perhaps we could trim them or explicitly allow them in the regex. Not sure why 0 padded numbers with an incomplete decimal are triggering a false alarm tho, I'll look into it.